### ISSUE TITLE
Validering av kodeverk

### ISSUE THREAD

**Issue by rockphotog** [HL7Norway/basisprofiler-r4#57]
Tester Torinox for validering (egen branch), og da var det ikke til å komme unna 

`[WARNING] Terminology service failed while validating code '' (system ''): Cannot retrieve valueset 'urn:oid:2.16.578.1.12.4.1.1.3402' (at Patient.address[0].extension[0].extension[0])`

Jeg ser ikke at vi får en fungerende terminologiserver basert på Healthterm og Volven-kodeverkene der med det første, og typiske kodeverk som kommunenummer er nok helt sikkert ikke pri -- kan det være mulighet å se på en enkel midlertidig løsning? Om enn lage et enkelt script som konverterer et Volven-kodeverk til et (CodeSystem+)ValueSet.

Nå som flere og flere tar i bruk f.eks. basisprofilene hadde det vært fint å kunne tilby mulighetene for 100% validering.

**Comment by rockphotog** [comment]
API for KLASS, som er kilden for 3402: https://data.ssb.no/api/klass/v1/api-guide.html

Eksempel `curl 'http://data.ssb.no/api/klass/v1/versions/1160' -i`

**Comment by thomiz** [comment]
En warning på en oid er i prinsippet ikke så viktig. "alle" forstår at en oid ikke kan resolves direkte i FHIR og at man derfor dessverre må behandle disse kodeverkene spesielt. 
Ønsker de som ønsker å validere mot kodene i kodeverket må i praksis ha tilgang til kodeverket og konstruere en måte å resolve oid'ene på en kodeverksserver. Det er kanskje mulig å definere opp et NamingSystem som beskriver for FHIR installasjonen hvordan kodeverket skal resolves lokalt. Da kan det også være en ide å bruke navnet på NamingSystem i lokale profiler forsåvidt.

**Comment by rockphotog** [comment]
Ja, "main issue" er egentlig at Volven er gammelt - og om vi bør gjøre noe ekstra for de Volven-kodeverk som finnes i basisprofiler og etter hvert områdeprofiler. 

En relativt enkel terminologiserver-fasade foran Volven på kort sikt burde vært kjekt, men skal det være driftsikkerhet på den må vi involvere NHN - og ta det som den del av grunnmursarbeidet. 

Kodeverk forvaltes fortsatt på Volven, selv om det er dratt inn i Healthterm (samtlige Volven-kodeverk er der samme kodesystem -- som nå da må være allerede utdatert).

**Comment by thomiz** [comment]
Enklere å lage statiske versjoner som distribueres via GitHub som hjelperessurser for validering av no-basis/områdeprofiler? Kan oppdateres med (ujevne) mellomrom når folk finner feil? 
Communityoppgave?

**Comment by rockphotog** [comment]
Ja, litt script-greier, så hvem som helst kan dra ned og kjøre et "oppdater-value-sett-rett-fra-ssb"-skript og commite nytt ValueSet - under en slags "tools"-fane, så det ikke er normativt i seg selv. Potensielt eget repo - kanskje til og med en hel terminologiserver for forking/nedlasting. Snowstorm er jo f.eks. bare en fork av HAPI med "nokå innhold".

**Comment by kennethmyhra** [comment]
Slik vi gjorde det for DocumentReference og Composition ved å opprette et [ValueSet](https://github.com/HL7Norway/basisprofiler-r4/blob/master/ValueSet/no-basis-documentreference-type.valueset.xml). Referer vi til ValueSet i StructureDefinition og System i ressursen.

Slik jeg forstår Thomas så er det dette som menes med statiske versjoner?

**Comment by rockphotog** [comment]
Note to self: ws.geonorge.no/kommuneinfo/v1/kommuner
